app/helpers/application_helper.rb

```diff
@@ -22,11 +22,10 @@ def inbox_count
   def notification_count
     return 0 unless user_signed_in?
 
-    count = Notification.for(current_user).where(new: true)
-    return nil if count.nil?
-    return nil unless count.count.positive?
+    count = Notification.for(current_user).where(new: true).pluck(:id).size
```

Comment:
Can we get Rails to generate a query that looks something like `SELECT COUNT(*) FROM \`notifications\` WHERE recipient_id = ?` instead of getting a list of IDs?

Reply:
I guess just something like `....where(...).count` would already do that.  Probably the `.nil?` check caused the extra queries.